src/loaddb/load_session.cpp

```diff
@@ -182,15 +184,21 @@ namespace cubload
 	    return;
 	  }
 
+        if (m_session.get_class_registry ().get_class_entry (m_batch.get_class_id ()) == NULL)
```

Comment:
We have to check on syntax_check flag as well, so I would like to propose following changes here.
* _if class is not registered_ and syntax check is enabled then return
* otherwise _if if class is not registered_ and syntax check is **not** enabled then `assert (false)`
```cpp
bool is_syntax_check_only = m_session.get_args ().syntax_check;
bool is_class_registered = m_session.get_class_registry ().get_class_entry (m_batch.get_class_id ()) != NULL;
if (!is_class_registered)
  {
    m_session.notify_batch_done (m_batch.get_id ());

    if (!is_syntax_check_only)
      {
        assert (false);
      }

    return;
  }
```